File: packages/runtime/nodejs/src/workflow/type/core/task.ts
import { WorkflowOutputs } from '../base';
import { IContext } from '.';

export interface ITask {
  id: string;
  processing: Promise<WorkflowOutputs>;
  context: IContext;
  cancel(): void;
}

export interface TaskParams {
  processing: Promise<WorkflowOutputs>;
  context: IContext;
}

File: packages/runtime-go/interface/runtime/task.go
/**
 * Copyright (c) 2025 Bytedance Ltd. and/or its affiliates
 * SPDX-License-Identifier: MIT
 */

package runtimeType

type ITask interface {
	GetID() string
	GetProcessing() <-chan WorkflowOutputs
	GetContext() IContext
	Cancel()
}

type TaskParams struct {
	Processing chan WorkflowOutputs
	Context    IContext
}
